### ISSUE TITLE
End support of nodejs v14

### ISSUE THREAD

**Issue by mshustov** [ClickHouse/clickhouse-js#148]
End-of-life nodejs v14.x is April 30, 2023. see https://github.com/nodejs/Release
We can remove v14 from the list of supported versions and CI checks.

**Comment by olexiyb** [comment]
In our project we need to use latest node 19(20) as it gives very noticeable performance improvements.
When we started to use node 19 we started to capture Timeout errors on slow queries (we did not set any timeouts). 
I was able to track down this to incorrect value in socket.timeout was set to 3000
![image](https://user-images.githubusercontent.com/352780/234502542-2332d555-9b5e-4b8a-8467-5890c07c1474.png)

I suspect this is somehow related to https://github.com/nodejs/node/issues/47137

After I have added
```
      request.once('socket', (socket) => {
        socket.setTimeout(this.config.request_timeout)
      })
```

The slow function started to work. This issue was reproducible on Mac M1 and alpine 3.17 (running in docker container)
